### PR TITLE
Add docs for SASL PLAIn support in Mirror Maker

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Documentation

### Description

This PR adds the documentation to the SASL PLAIN support in Kafka Mirror Maker deployment.

### Checklist

- [x] Update documentation

### CODE REVIEW

File: documentation/book/con-kafka-connect-authentication.adoc [hunk 1]
```diff
@@ -74,10 +74,13 @@ spec:
 To configure Kafka Connect to use SASL based PLAIN authentication, set the `type` property to `plain`.
 This authentication mechanism requires a username and password.
 
+WARNING: SASL PLAIN mechanism transfers the username and password in cleartext over the network.
+It should not be used without TLS encryption.
```
Comment (ppatierno) [inline documentation/book/con-kafka-connect-authentication.adoc]: I would say in the other way around, "it should be used only with TLS encryption"

File: documentation/book/con-kafka-mirror-maker-authentication.adoc [hunk 2]
```diff
@@ -87,3 +89,45 @@ spec:
         password: my-producer-password-key
   # ...
 ----
+
+== PLAIN authentication
+
+To configure Kafka Mirror Maker to use PLAIN authentication, set the `type` property to `plain`.
+The broker listener to which clients are connecting must also be configured to use PLAIN SASL authentication.
```
Comment (ppatierno) [inline documentation/book/con-kafka-mirror-maker-authentication.adoc]: "SASL PLAIN" not "PLAIN SASL"

File: documentation/book/con-kafka-mirror-maker-authentication.adoc [hunk 3]
```diff
@@ -87,3 +89,45 @@ spec:
...
+This authentication mechanism requires a username and password.
+
+WARNING: SASL PLAIN mechanism transfers the username and password in cleartext over the network.
+It should not be used without TLS encryption.
```
Comment (ppatierno) [inline documentation/book/con-kafka-mirror-maker-authentication.adoc]: Ditto as above using the other way around

File: documentation/book/con-kafka-connect-authentication.adoc [hunk 4]
```diff
@@ -74,10 +74,13 @@ spec:
 To configure Kafka Connect to use SASL based PLAIN authentication, set the `type` property to `plain`.
```
Comment (laidan6000) [inline documentation/book/con-kafka-connect-authentication.adoc]: Use a hyphen for a compound adjective: SASL-based PLAIN authentication

File: documentation/book/con-kafka-connect-authentication.adoc [hunk 5]
```diff
@@ -74,10 +74,13 @@ spec:
...
 This authentication mechanism requires a username and password.
 
+WARNING: SASL PLAIN mechanism transfers the username and password in cleartext over the network.
```
Comment (laidan6000) [inline documentation/book/con-kafka-connect-authentication.adoc]:
```suggestion
WARNING: The SASL PLAIN mechanism involves transferring the username and password across the network in cleartext. Only use SASL-Plain authentication if TLS encryption is enabled.
```

File: documentation/book/con-kafka-mirror-maker-authentication.adoc [hunk 6]
```diff
@@ -11,7 +11,9 @@ The currently supported authentication types are:
 
 * TLS client authentication
 * SASL based authentication using the SCRAM-SHA-512 mechanism
+* SASL based authentication using the PLAIN mechanism
```
Comment (laidan6000) [inline documentation/book/con-kafka-mirror-maker-authentication.adoc]: SASL-based authentication

File: documentation/book/con-kafka-mirror-maker-authentication.adoc [hunk 7]
```diff
@@ -87,3 +89,45 @@ spec:
         password: my-producer-password-key
   # ...
 ----
+
+== PLAIN authentication
+
+To configure Kafka Mirror Maker to use PLAIN authentication, set the `type` property to `plain`.
+The broker listener to which clients are connecting must also be configured to use SASL PLAIN authentication.
```
Comment (laidan6000) [inline documentation/book/con-kafka-mirror-maker-authentication.adoc]:
```suggestion
The broker listener to which clients will connect must also be configured to use SASL PLAIN authentication.
```

File: documentation/book/con-kafka-mirror-maker-authentication.adoc [hunk 8]
```diff
@@ -87,3 +89,45 @@ spec:
...
+This authentication mechanism requires a username and password.
+
+WARNING: SASL PLAIN mechanism transfers the username and password in cleartext over the network.
```
Comment (laidan6000) [inline documentation/book/con-kafka-mirror-maker-authentication.adoc]: Please repeat the earlier text for the warning.

File: documentation/book/con-kafka-connect-authentication.adoc [hunk 9]
```diff
@@ -71,13 +71,16 @@ spec:
 
 == SASL based PLAIN authentication
 
-To configure Kafka Connect to use SASL based PLAIN authentication, set the `type` property to `plain`.
+To configure Kafka Connect to use SASL-based PLAIN authentication, set the `type` property to `plain`.
 This authentication mechanism requires a username and password.
 
+WARNING: The SASL PLAIN mechanism involves transferring the username and password across the network in cleartext.
```
Comment (PaulRMellor) [inline documentation/book/con-kafka-connect-authentication.adoc]:
```suggestion
WARNING: The SASL PLAIN mechanism will transfer the username and password across the network in cleartext.
```

File: documentation/book/con-kafka-mirror-maker-authentication.adoc [hunk 10]
```diff
@@ -87,3 +89,45 @@ spec:
...
   # ...
 ----
+
+== PLAIN authentication
+
+To configure Kafka Mirror Maker to use PLAIN authentication, set the `type` property to `plain`.
+The broker listener to which clients will connect must also be configured to use SASL PLAIN authentication.
+This authentication mechanism requires a username and password.
+
+WARNING: The SASL PLAIN mechanism involves transferring the username and password across the network in cleartext.
```
Comment (PaulRMellor) [inline documentation/book/con-kafka-mirror-maker-authentication.adoc]:
```suggestion
WARNING: The SASL PLAIN mechanism will transfer the username and password across the network in cleartext.
```

**Pull request merged**